src/WebJobs.Extensions.DurableTask/LocalHttpListener.cs

```diff
@@ -63,27 +81,25 @@ public async Task StopAsync()
             this.IsListening = false;
         }
 
-        private IWebHost CreateWebHost(Uri listenUri)
+        private int GetAvailablePort()
         {
-            if (listenUri == null)
+            // If we are able to successfully start a listener looking on the default port without 
+            // an exception, we can use the default port. Otherwise, let the TcpListener class decide for us.
+            try
             {
-                throw new ArgumentNullException(nameof(listenUri));
+                var listener = new TcpListener(IPAddress.Loopback, DefaultPort);
+                listener.Start();
+                listener.Stop();
+                return DefaultPort;
             }
-
-            if (listenUri.AbsolutePath.Length > 1)
+            catch (SocketException)
             {
-                throw new ArgumentException($"The listen URL must not contain a path.", nameof(listenUri));
+                var listener = new TcpListener(IPAddress.Loopback, 0);
```

Comment:
It might be worth adding a comment with this URL: https://stackoverflow.com/a/150974/2069 :)